src/main/java/io/inventi/shiro/api/realm/service/UserService.java

```diff
@@ -0,0 +1,68 @@
+package io.inventi.shiro.api.realm.service;
+
+import io.inventi.shiro.api.realm.domain.User;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.web.client.RestTemplateBuilder;
+import org.springframework.cache.annotation.CacheEvict;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+public class UserService {
+
+    private static final Logger logger = LoggerFactory.getLogger(UserService.class);
+
+    private RestTemplate userApiTemplate;
+
+    public UserService(String rootEndpoint, RestTemplateBuilder restTemplateBuilder) {
+
+        userApiTemplate = restTemplateBuilder
+                .rootUri(rootEndpoint)
+                .build();
+    }
+
+    public Optional<User> findUser(String userId) {
```

Comment:
Method names are confusing. findUser and getUser by method names seem to do the same thing